### PR TITLE
kfam: prometheus monitoring integration

### PR DESCRIPTION
**Pull request by kunmingg**
requestCounter: 
normal request count with tags: `COMPONENT, KIND, REQUSER, ACTION, PATH`

requestErrorCounter:
failed request count with tags: `COMPONENT, KIND, REQUSER, ACTION, PATH, SEVERITY`

serviceHeartbeat:
counter that increase value by 1 every 10 seconds, indicating the kfam backend is alive.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/4674)

### CODE REVIEW

File: components/access-management/kfam/monitoring.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
+package kfam
```
Comment (zhenghuiwang) [inline components/access-management/kfam/monitoring.go]: Do we expect other KF components to follow this pattern(same set of counter names, types etc)? If so, can it be put as a library outside `kfam`?

Comment (kunmingg) [reply components/access-management/kfam/monitoring.go]: We can put that effort into v1.0.1 and avoid blocking release of v1.0.0?
@jlewi

**Comment by kunmingg** [comment]
related: https://github.com/kubeflow/kubeflow/issues/4076

File: components/access-management/kfam/monitoring.go (hunk 1 above)
Comment (jlewi) [reply components/access-management/kfam/monitoring.go]: +1 I suspect we will likely need to do some more refactoring post 1.0 as we will identify missing metrics etc... so I would be in favor of getting this merged now and then taking a more holistic look at our monitoring.

**Pull request merged**